Sources/PackageRegistry/RegistryManager.swift

```diff
@@ -0,0 +1,296 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import PackageLoading
+import PackageModel
+
+import TSCBasic
+import TSCUtility
+
+import struct Foundation.URL
+import struct Foundation.URLComponents
+import struct Foundation.URLQueryItem
+
+import Dispatch
+
+public enum RegistryError: Error {
+    case invalidOperation
+    case invalidResponse
+    case invalidURL
+    case invalidChecksum(expected: String, actual: String)
+}
+
+fileprivate let userAgent = "SwiftPackageManager/\(ToolsVersion.currentToolsVersion)"
+
+public final class RegistryManager {
+    internal static var clientFactory: () -> HTTPClientProtocol = {
+        var configuration = HTTPClientConfiguration()
+        configuration.followRedirects = false
+
+        return HTTPClient(configuration: configuration, handler: nil, diagnosticsEngine: nil)
```

Comment:
we would want to pass in a diagnosticsEngine here, so maybe clientFactory should take one, etc

Reply:
Added with 15c90e065c15133a953c40d30b3083dd6d3081da. I started add diagnostics engine parameters to `getContainer`, but I quickly got outside my comfort zone for this PR. Something to consider in a follow-up /cc @abertelrud